packages/runtime/src/widgets/__tests__/Conditional.test.tsx

```diff
@@ -0,0 +1,233 @@
+import { render } from "@testing-library/react";
+import {
+  Conditional,
+  ConditionalProps,
+  hasProperStructure,
+  extractWidget,
+  extractCondition,
+} from "../Conditional";
+
+describe("Conditional Component", () => {
+  test('renders the widget when "if" condition is met', () => {
+    const conditionalProps: ConditionalProps = {
+      conditions: [
+        {
+          if: "true",
+          Text: { name: "Text", properties: { text: "Widget A" } },
+        },
+        {
+          else: null,
+          Text: { name: "Text", properties: { text: "Widget B" } },
+        },
+      ],
+    };
+    expect(() =>
+      render(<Conditional conditions={conditionalProps.conditions} />)
+    ).not.toThrow();
+  });
+
+  test('renders the widget when "elseif" condition is met', () => {
+    const conditionalProps: ConditionalProps = {
+      conditions: [
+        {
+          if: "false",
+          Text: { name: "Text", properties: { text: "Widget A" } },
+        },
+        {
+          elseif: "true",
+          Text: { name: "Text", properties: { text: "Widget B" } },
+        },
+      ],
+    };
+
+    expect(() =>
+      render(<Conditional conditions={conditionalProps.conditions} />)
+    ).not.toThrow();
```

Comment:
@anserwaseem take a look at the EnsembleApp.test.tsx file. You can assert that the elements you expect are rendered.